packages/runtime/src/templates/requireHooks.ts

```diff
@@ -11,86 +11,89 @@ const resolveFilename = (mod as any)._resolveFilename
+          ['react-dom/server.browser', `next/dist/compiled/react-dom/server.browser`],
+          ['react-dom/server.edge', `next/dist/compiled/react-dom/server.edge`],
+          ['react-server-dom-webpack/client', `next/dist/compiled/react-server-dom-webpack/client`],
+          ['react-server-dom-webpack/client.edge', `next/dist/compiled/react-server-dom-webpack/client.edge`],
+          ['react-server-dom-webpack/server.edge', `next/dist/compiled/react-server-dom-webpack/server.edge`],
+          ['react-server-dom-webpack/server.node', `next/dist/compiled/react-server-dom-webpack/server.node`],
+          ['styled-jsx', 'styled-jsx'],
+          ['styled-jsx/style', 'styled-jsx/style'],
+        ]),
+      )
+    }
   }
+}
 
-  if (config.experimental.serverActions) {
-    requireHooks.set(
-      'experimental',
-      new Map([
-        ['react', require.resolve(`next/dist/compiled/react-experimental`, opts)],
-        ['react/jsx-runtime', require.resolve(`next/dist/compiled/react-experimental/jsx-runtime`, opts)],
-        ['react/jsx-dev-runtime', require.resolve(`next/dist/compiled/react-experimental/jsx-dev-runtime`, opts)],
-        ['react-dom', require.resolve(`next/dist/compiled/react-dom-experimental/server-rendering-stub`, opts)],
-        ['react-dom/client', require.resolve(`next/dist/compiled/react-dom-experimental/client`, opts)],
-        ['react-dom/server', require.resolve(`next/dist/compiled/react-dom-experimental/server`, opts)],
-        ['react-dom/server.browser', require.resolve(`next/dist/compiled/react-dom-experimental/server.browser`, opts)],
-        ['react-dom/server.edge', require.resolve(`next/dist/compiled/react-dom-experimental/server.edge`, opts)],
-        [
-          'react-server-dom-webpack/client',
-          require.resolve(`next/dist/compiled/react-server-dom-webpack-experimental/client`, opts),
-        ],
-        [
-          'react-server-dom-webpack/client.edge',
-          require.resolve(`next/dist/compiled/react-server-dom-webpack-experimental/client.edge`, opts),
-        ],
-        [
-          'react-server-dom-webpack/server.edge',
-          require.resolve(`next/dist/compiled/react-server-dom-webpack-experimental/server.edge`, opts),
-        ],
-        [
-          'react-server-dom-webpack/server.node',
-          require.resolve(`next/dist/compiled/react-server-dom-webpack-experimental/server.node`, opts),
-        ],
-        ['styled-jsx', require.resolve('styled-jsx', opts)],
-        ['styled-jsx/style', require.resolve('styled-jsx/style', opts)],
-      ]),
-    )
-  }
+const resolveRequireHooks = () => {
+  // we may have changed the working directory in the handler
+  const opts = { paths: [process.cwd()] }
+
+  // resolve require hooks with module paths
+  requireHooks.forEach((mode) => {
+    mode.forEach((hook, path) => {
+      try {
+        const resolvedPath = require.resolve(path, opts)
+        mode.set(hook, resolvedPath)
+      } catch {
+        // ignore
```

Comment:
Do we need any logging here for our observability or is it safe to just ignore?

Reply:
hey @MarcL, this is safe to ignore here because it's not actually an error - we want to delete the hook in this case because it happens when a module isn't found, i.e. an older version of Next
I have updated with a better comment, thanks